torch/nn/_functions/vision.py

```diff
@@ -93,35 +93,74 @@ def _enforce_cudnn(input):
     @staticmethod
     def forward(ctx, theta, size):
         assert type(size) == torch.Size
-        N, C, H, W = size
-        ctx.size = size
-        if theta.is_cuda:
-            AffineGridGenerator._enforce_cudnn(theta)
-            assert False
-        ctx.is_cuda = False
-        base_grid = theta.new(N, H, W, 3)
-        linear_points = torch.linspace(-1, 1, W) if W > 1 else torch.Tensor([-1])
-        base_grid[:, :, :, 0] = torch.ger(torch.ones(H), linear_points).expand_as(base_grid[:, :, :, 0])
-        linear_points = torch.linspace(-1, 1, H) if H > 1 else torch.Tensor([-1])
-        base_grid[:, :, :, 1] = torch.ger(linear_points, torch.ones(W)).expand_as(base_grid[:, :, :, 1])
-        base_grid[:, :, :, 2] = 1
-        ctx.base_grid = base_grid
-        grid = torch.bmm(base_grid.view(N, H * W, 3), theta.transpose(1, 2))
```

Comment:
This implementation seems cleaner to me than the original, but I have not done performance testing. Should I backport this style to the 2d case?